core/src/main/java/hudson/model/Computer.java

```diff
@@ -620,7 +622,8 @@ public BuildTimelineWidget getTimeline() {
     @Exported
     @Override
     public boolean isOffline() {
-        return temporarilyOffline || getChannel() == null;
+        var node = getNode();
+        return node == null || node.isTemporarilyOffline() || getChannel() == null;
```

Comment:
I think you meant
```suggestion
        var node = getNode();
        return node != null && node.isTemporarilyOffline() || getChannel() == null;
```
or more simply
```suggestion
        return isTemporarilyOffline() || getChannel() == null;
```
If the node was deleted but the computer is still connected and busy, it should not be marked as offline.